scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -774,29 +774,33 @@ def test_input_types(self):
     @pytest.mark.timeout(120)  # prerelease_deps_coverage_64bit_blas job
     @sup_sparse_efficiency
     def test_random(self):
-        def random_triangle_matrix(n, lower=True):
-            A = scipy.sparse.random(n, n, density=0.1, format='coo')
+        def random_triangle_matrix(n, lower=True, format="csr"):
+            A = scipy.sparse.random(n, n, density=0.1, format='lil')
             if lower:
-                A = scipy.sparse.tril(A)
+                A = scipy.sparse.tril(A,format="lil")
             else:
-                A = scipy.sparse.triu(A)
-            A = A.tocsr(copy=False)
+                A = scipy.sparse.triu(A,format="lil")
             for i in range(n):
                 A[i, i] = np.random.rand() + 1
+            if format=="csc":
+                A = A.tocsc(copy=False)
+            else:
+                A = A.tocsr(copy=False)
             return A
 
         np.random.seed(1234)
         for lower in (True, False):
-            for n in (10, 10**2, 10**3):
-                A = random_triangle_matrix(n, lower=lower)
-                for m in (1, 10):
-                    for b in (np.random.rand(n, m),
-                              np.random.randint(-9, 9, (n, m)),
-                              np.random.randint(-9, 9, (n, m)) +
-                              np.random.randint(-9, 9, (n, m)) * 1j):
-                        x = spsolve_triangular(A, b, lower=lower)
-                        assert_array_almost_equal(A.dot(x), b)
-                        x = spsolve_triangular(A, b, lower=lower,
-                                               unit_diagonal=True)
-                        A.setdiag(1)
-                        assert_array_almost_equal(A.dot(x), b)
+            for format in ["csr", "csc"]:
+                for n in (10, 10**2, 10**3):
+                    A = random_triangle_matrix(n, lower=lower)
```

Comment:
I am always a fan of pytest parametrization for such tests as it makes debugging much easier in my opinion. Could you adapt this code? I think it is very much historically grown.

Reply:
Thanks for suggesting this! e645e4cc799141efe9a387a8f280f690f10ce462 makes the test parametric, removing the nested loops in a single testcase. Additionally, in 2f6802eacc6f357728be34ce7cd2e06f0be5d8ce, I've added tests for complex-valued matrices.